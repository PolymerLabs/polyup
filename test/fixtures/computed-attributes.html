<link rel='import' href='../polymer/polymer.html'>

<polymer-element name='computed-attributes-test-elem'>
  <template>
    <div id='{{x}}'></div>
    <div id='{{x + x}}'></div>
    <div class='big class{{x}}'></div>
    <span>{{x + x}}</span>
    <template if='{{x}}'>
      Hi hello how {{x * x}} are you.
    </template>
  </template>
  <script>
    Polymer('computed-attributes-test-elem', {
      x: 1,
      computed: {
        'y': 'x + x',
        z: 'double(x)',
        zz: 'double(x) + double(x)',
        'identity': 'x'
      },
      double: function(val) {
        return val * val;
      }
    });
  </script>
</polymer-element>
