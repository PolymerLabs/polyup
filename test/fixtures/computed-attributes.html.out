<link rel="import" href="../polymer/polymer.html">

<!-- This element tests out migrating computed attributes. -->
<dom-module id="computed-attributes-test-elem">
  <template>
    <div id="{{x}}"></div>
    <div id="{{_computeY(x)}}"></div>
    <div class$="{{_computeClass(x)}}"></div>
    <span>{{_computeY(x)}}</span>
    <template is="dom-if" if="{{x}}">
      Hi hello how <span>{{_computeExpression2(x)}}</span> are you.
    </template>
  </template>
  <script>
    Polymer({
      is: 'computed-attributes-test-elem',
      properties: {
        identity: { computed: 'x' },
        x: {
          type: Number,
          value: 1
        },
        y: { computed: '_computeY(x)' },
        z: { computed: 'double(x)' },
        zz: { computed: '_computeZz(x)' }
      },
      double: function (val) {
        return val * val;
      },
      _computeY: function (x) {
        return x + x;
      },
      _computeClass: function (x) {
        return 'big class' + x;
      },
      _computeExpression2: function (x) {
        return x * x;
      },
      _computeZz: function (x) {
        return this.double(x) + this.double(x);
      }
    });
  </script>
</dom-module>
